plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConfig.java

```diff
@@ -70,7 +70,7 @@
     private boolean singleStatementWritesOnly;
 
     private DataSize maxSplitSize = DataSize.of(64, MEGABYTE);
-    private int maxPartitionsPerScan = 100_000;
+    private int maxPartitionsPerScan = 1_000_000;
```

Comment:
Won't it impact the memory. IIRC this check was added to restrict the memory usage by `HivePartition` which internally has a `Map` for the partition keys. 

cc: @findepi / @electrum

Reply:
The `Avoid loading partitions eagerly in HiveSplitManager` improves split enumeration to avoid parsing / loading partitions eagerly. The memory should no longer be of concern. The property now is mostly for system administrators to prevent unnecessary large scans.